src/hist/basehist.py

```diff
@@ -301,7 +301,8 @@ def plot(self, *args: Any, **kwargs: Any) -> "Union[Hist1DArtists, Hist2DArtists
         """
         Plot method for BaseHist object.
         """
-        if self.ndim == 1:
+        len_cat = np.sum([ax.traits.discrete for ax in self.axes])
+        if self.ndim == 1 or (self.ndim == 2 and len_cat == 1):
```

Comment:
I'm not sure I like this. It's also valid to make a 2D plot where one axis is categorical (or two, for that matter). I think it would be more predictable if `.plot()` on a 2D histogram always made a 2D plot. But I could be convinced. What do other people think? @nsmith-, @matthewfeickert, perhaps?

Reply:
What @nsmith- 's coffea API does is have overlay keyword to control this for non-discrete axes, but if think the number of 2d histograms of type (sample, quantity) is gonna be higher than (sample, sample) at least if one of the axes is StrCat (or just Cat)

Reply:
I suppose in coffea its either `plot1d` or `plot2d`. But indeed if `overlay: Optional[str]` is a parameter, one can take that to decide if a 2D hist is a 2D plot or a set of 1D plots with the overlay axis as specified.

Reply:
Another option might be to follow pandas style, where the hist would have to be turned into some kind of `HistGroupBy` object before plotting, e.g. `h.groupby("region").plot()`.

Reply:
tbh since `hist` does `plot1d()` and `plot2d()` as well and `plot()` is just a two char shortcut, this is a trifle.

@nsmith- can you think of any other use for some kind of groupby fcn than this? I don't think it's worth fiddling with just for this

Reply:
Not yet, probably you are right to leave it as-is for now.